lib/Demangling/Demangler.cpp

```diff
@@ -783,7 +782,22 @@ NodePointer Demangler::demangleOperator() {
     case 'V': return demangleAnyGenericType(Node::Kind::Structure);
     case 'W': return demangleWitness();
     case 'X': return demangleSpecialType();
-    case 'Y': return createNode(Node::Kind::AsyncAnnotation);
+    case 'Y':
+      switch (char c2 = nextChar()) {
+      case 'a':
+        return createNode(Node::Kind::AsyncAnnotation);
+      case 'b':
+        return createNode(Node::Kind::ConcurrentFunctionType);
+      case 'j':
+        return demangleDifferentiableFunctionType();
+      case 'k':
+        return createType(
+            createWithChild(Node::Kind::NoDerivative, popTypeAndGetChild()));
+      default:
+        pushBack();
+        pushBack();
```

Comment:
Why this? I think it should just return null in case there is an invalid 'Y' operator.

Reply:
This is mirroring line 765 where sub-operators of 'H' are being parsed. I didn't understand why it existed there, so I copied the behavior just in case. I've removed it now since you requested it.

Reply:
yeah, also the 'H' demangling should just return a nullptr.